Synview-frontend/src/components/SyncForm.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from "react";
+import { useGetMyCommitsMutation } from "../services/apiSlice.ts";
+import { useAppSelector } from "../hooks.ts";
+
+export default function SyncForm() {
+  const [githubUsername, setGithubUsername] = useState("");
+  const [projectName, setProjectName] = useState("");
+  const [getCommits] = useGetMyCommitsMutation();
+  const project_id = useAppSelector((state) => state.githubModal.project_id);
+  const user_id = useAppSelector((state) => state.githubModal.user_id);
+
+  const handleSyncForm = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    try {
+      if (projectName && githubUsername && project_id && user_id) {
+        await getCommits({
+          project_id: project_id,
+          user_id: user_id,
+          repo_name: projectName,
+          github_user: githubUsername,
+        });
+        setGithubUsername("");
+        setProjectName("");
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  return (
+    <div>
+      <fieldset className="fieldset border-1 p-4 rounded-box border-neutral-600 bg-neutral-800">
+        <div className="flex flex-col justify-center">
+          <form onSubmit={handleSyncForm}>
+            <label>Github username</label>
```

Comment:
You should set `htmlFor` on the label to the `input`'s `id`. This makes it so users can click the label, improves accessibility, and typically makes testing easier.